### PR TITLE
Modify S6703: Remove var substitution FPs APPSEC-1100

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
This PR fixes False Positives on placeholders resembling `{sonarloris}`. It also adds clarity in the patternNot list

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml [hunk 1]
```diff
@@ -183,8 +183,22 @@ provider:
...
-            - "^(?i)pass(word|wd)?(?-i)$|^(?i)pwd(?-i)$|^\\[?PLACEHOLDER\\]?$|^%s$|^\\*{3,}$|^\\{\\d*\\}$|^\\#\\{|^\\{{2,}"
+            - "^(?i)pass(word|wd)$"
+            - "^(?i)pwd"
+            - "^\\[?PLACEHOLDER\\]?$"
+            - "^%s$"
+            - "^\\*{3,}$"
+            - "^\\{\\d*\\}$"
+            - "^\\#\\{"
+            - "^\\{{2,}"
+            - "^\\{+[^}]*\\}++$"
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:194]:
```suggestion
            # Support full and partial substitution
            - "\\{+[^}]*\\}++"
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml [hunk 2]
```diff
@@ -183,8 +183,23 @@ provider:
...
+            - "^(?i)pass(word|wd)$"
+            - "^(?i)pwd"
+            - "^\\[?PLACEHOLDER\\]?$"
+            - "^%s$"
+            - "^\\*{3,}$"
+            - "^\\{\\d*\\}$"
+            - "^\\#\\{"
+            - "^\\{{2,}"
+            # Support for full and partial substitution
+            - "\\{+[^}]*\\}++"
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:195]:
```suggestion
            - "\\{+[^}]*\\}++"
            
            # Azure RM templates
            - "parameters\\(+[^)]*\\)++"
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml [hunk 3]
```diff
@@ -183,8 +183,23 @@ provider:
...
       examples:
+        - text: |
+              "ConnectionStrings": {
+                "DefaultConnection": "Server=tcp:url.windows.example.com,1433;Initial Catalog=sql-example;Persist Security Info=False;User ID={EXAMPLE_USER};Password={PASSWORD_HERE};MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=60;"
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:199]:
```suggestion
                "DefaultConnection": "Server=tcp:url.windows.example.com,1433;Initial Catalog=sql-example;Persist Security Info=False;User ID={EXAMPLE_USER};Password={PASSWORD_HERE};MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=60;"
                "sqlConnectionString": "concat('Data Source=',parameters('a'),';Initial Catalog=',parameters('b'),';User ID=',parameters('c'),';Password=',parameters('d'))"
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml [hunk 4]
```diff
@@ -183,8 +183,26 @@ provider:
         post:
           # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
           patternNot:
-            - "^(?i)pass(word|wd)?(?-i)$|^(?i)pwd(?-i)$|^\\[?PLACEHOLDER\\]?$|^%s$|^\\*{3,}$|^\\{\\d*\\}$|^\\#\\{|^\\{{2,}"
+            - "^(?i)pass(word|wd)$"
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:186]: word or wd are not optional anymore. Is this change intended?

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml [hunk 5]
```diff
@@ -183,8 +183,26 @@ provider:
...
+            - "^(?i)pwd"
+            - "^\\[?PLACEHOLDER\\]?$"
+            - "^%s$"
+            - "^\\*{3,}$"
+            - "^\\{\\d*\\}$"
+            - "^\\#\\{"
+            - "^\\{{2,}"
+            # Support for full and partial substitution
+            - "\\{+[^}]*\\}++"
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:195]: Why is the closing curly bracket matched possessively? I think it would only make sense to match the part in the middle possessively:
```suggestion
            - "\\{+[^}]++\\}+"
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml [hunk 6]
```diff
@@ -183,8 +183,26 @@ provider:
         post:
           # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
           patternNot:
-            - "^(?i)pass(word|wd)?(?-i)$|^(?i)pwd(?-i)$|^\\[?PLACEHOLDER\\]?$|^%s$|^\\*{3,}$|^\\{\\d*\\}$|^\\#\\{|^\\{{2,}"
+            - "^(?i)pass(word|wd)$"
+            - "^(?i)pwd"
+            - "^\\[?PLACEHOLDER\\]?$"
+            - "^%s$"
+            - "^\\*{3,}$"
+            - "^\\{\\d*\\}$"
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:191]: This can be removed if you add the new regexp:
```\\{+[^}]*\\}++```

**Review by daniel-teuchert-sonarsource — CHANGES_REQUESTED** [review]
Some things that may need a change.
Also, while you are at it, maybe add the examples from RSPEC to the tests?

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml (hunk 4 above)
Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:186]: correct !
```suggestion
            - "^(?i)pass(word|wd)?$"
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml (hunk 6 above)
Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/odbc.yaml:191]:
```suggestion
```

**Review by daniel-teuchert-sonarsource — APPROVED** [review]
LGTM!
